script/vsts/upload-artifacts.js

```diff
@@ -65,27 +65,44 @@ async function uploadArtifacts() {
     return;
   }
 
-  console.log(
-    `Uploading ${
-      assets.length
-    } release assets for ${releaseVersion} to S3 under '${bucketPath}'`
-  );
-
-  await uploadToS3(
-    process.env.ATOM_RELEASES_S3_KEY,
-    process.env.ATOM_RELEASES_S3_SECRET,
-    process.env.ATOM_RELEASES_S3_BUCKET,
-    bucketPath,
-    assets
-  );
+  if (
+    process.env.ATOM_RELEASES_S3_KEY &&
+    process.env.ATOM_RELEASES_S3_SECRET
```

Comment:
Why is bucket missing here?
```suggestion
    process.env.ATOM_RELEASES_S3_SECRET &&
    process.env.ATOM_RELEASES_S3_BUCKET
```

Reply:
I figure if you haven't set a secret or an API key you're not going to be able to upload, and I was honestly just trying to stop adding so many lines to check.

If you want to add it, you can. We will have set the others to be empty, so it will skip. Good enough, right? Whatever you like, you can do.

I am really not pleased we need to set five variables empty in CI now. But I mean if this is what it takes to move forward, then okay.

Reply:
If an environment variable is not defined it will be `undefined`. You don't need to set it to empty. 

https://howtodoinjava.com/typescript/truthy-and-falsy/#:~:text=%20%20%20%20Falsy%20Value%2FExpression%20%20,to%20false%20...%20%202%20more%20rows%20

Reply:
For the record, I do not understand how this would behave if you have an invalid string as the S3 bucket.

Probably upload to a bucket called literally `$(ATOM_RELEASES_S3_BUCKET)`.

Reply:
Is S3 bucket optional or it required for uploading?

Reply:
It is passed to the "upload to s3" lib, so I'm going to assume if it's some really invalid value it might error out on attempting to upload to S3.

Please read script/vsts/lib/upload-to-s3.js, I don't remember what it does with that var.

The lib handles the value, and I think it's reasonable to expect it doesn't handle empty or incorrect values, it just hits "send" off to s3 and sees if it works.

A bucket is just like a folder in your Dropbox/Google Drive or whatever.

Reply:
The bucket is optional, but Atom probably uses it. So we will check for this too, just in case. Later when we upstreamed this, we can adjust it if they said they don't set this.
https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html

Reply:
Here's the module it uses: https://github.com/aws/aws-sdk-js

Reply:
I have explained this multiple times.

Azure DevOps does not leave the env vars undefined. It sets them explicitly to this string: `'$(VAR_NAME_HERE)'`.

Yes this is counter-intuitive and unexpected, but it is a documented feature of DevOps.

Do not misunderstand me or skip my explanations, I have tried this all the ways and confirmed it to be the case. I have linked to the docs. I know what I'm talking about.

Reply:
Try it and see.